src/shared/components/cosmic/CosmicMutationTable.tsx

```diff
@@ -0,0 +1,95 @@
+import {CosmicMutation} from "../../api/CBioPortalAPIInternal";
+import EnhancedReactTable from "../enhancedReactTable/EnhancedReactTable";
+import {IEnhancedReactTableProps} from "../enhancedReactTable/IEnhancedReactTableProps";
+import ProteinChangeColumnFormatter from "../mutationTable/column/ProteinChangeColumnFormatter";
+
+
+export interface ICosmicTableProps extends IEnhancedReactTableProps<CosmicMutation> {}
+
+// EnhancedReactTable is a generic component which requires data type argument
+class ReactTable extends EnhancedReactTable<CosmicMutation> {}
+
+/**
+ * @author Selcuk Onur Sumer
+ */
+export default class CosmicMutationTable extends React.Component<ICosmicTableProps, {}>
+{
+
+    constructor(props: ICosmicTableProps)
+    {
+        super(props);
+        this.state = {};
+    }
+
+    public render()
+    {
+        const {
+            reactTableProps,
+            initItemsPerPage,
+            headerControlsProps,
+            columns,
+            rawData
+        } = this.mergeProps(this.props);
+
+        return(
+            <div>
+                <ReactTable
+                    reactTableProps={reactTableProps}
+                    initItemsPerPage={initItemsPerPage}
+                    headerControlsProps={headerControlsProps}
+                    columns={columns}
+                    rawData={rawData}
+                />
+            </div>
+        );
+    }
+
+    private mergeProps(props:ICosmicTableProps):ICosmicTableProps
+    {
+        const columns = {
+            cosmicId: {
+                name: "COSMIC ID",
+                priority: 1.00,
+                dataField: "cosmicMutationId",
+                sortable: true
+            },
+            proteinChange: {
+                name: "Protein Change",
+                priority: 2.00,
+                dataField: "proteinChange",
+                sortable: ProteinChangeColumnFormatter.sortFunction,
+                filterable: true
+            },
+            occurrence: {
+                name: "Occurrence",
+                priority: 3.00,
+                dataField: "count",
+                sortable: true
+            }
+        };
+
+        const defaultProps:ICosmicTableProps = {
+            rawData: [],
+            columns: columns,
+            initItemsPerPage: 10,
+            headerControlsProps: {
+                showPagination: true,
+                showHideShowColumnButton: false,
+                showCopyAndDownload: false,
+                showSearch: false,
+                copyDownloadClassName: "pull-right",
+                searchClassName: "pull-left",
+            },
+            reactTableProps: {
+                className: "table table-striped table-border-top",
+                hideFilterInput:true,
+                defaultSort: {column: columns.occurrence.name, direction: 'desc'}
+            }
+        };
+
+        // merge provided props with the default props
+        return lodash.merge(defaultProps, props);
```

Comment:
i think we should have convention of lodash = _   it's used that most everywhere else in project